### PR TITLE
Fix Entity Type

### PR DESCRIPTION
**Pull request by Alfred-Mountfield**
## 🌟 What is the purpose of this PR?

The EntityType struct accidentally didn't let you set a property as being an array. This fixes that

## 🛡 What tests cover this?

- The full test suite, the `Book` test entity has been updated to have an array

## ❓ How to test this?

1. Follow the README as normal

### CODE REVIEW

File: packages/graph/hash_graph/lib/graph/src/ontology/types/serde_shared/array.rs [hunk 1]
```diff
@@ -95,4 +115,29 @@ mod tests {
...
+
+    mod value_or_array {
+        use serde_json::json;
+
+        use super::*;
+        use crate::ontology::types::serde_shared::tests::{check, StringTypeStruct};
+
+        #[test]
+        fn value() -> Result<(), serde_json::Error> {
+            check(&ValueOrArray::Value("value".to_owned()), json!("value"))
```
Comment (thehabbos007) [inline packages/graph/hash_graph/lib/graph/src/ontology/types/serde_shared/array.rs:127]: nit:
```suggestion
            check(
                &ValueOrArray::Value(StringTypeStruct::default()),
                json!({
                    "type": "string"
                }),
            )
```
would be more consistent with the rest of the tests

**Review by thehabbos007 — DISMISSED** [review]
I approve of the changes, but have a nit

File: packages/graph/hash_graph/lib/graph/src/ontology/types/serde_shared/array.rs (hunk 1 above)
Comment (Alfred-Mountfield) [reply packages/graph/hash_graph/lib/graph/src/ontology/types/serde_shared/array.rs:127]: rustfmt I'm afraid

**Comment by thehabbos007** [comment]
Just something I remembered which might be worth just drive-by adding here:

The Entity Type JSON Schema doesn't have the current `$ref` structure for Links
https://github.com/hashintel/hash/blob/afa24c0090a82441d61e51a0beb7442c08ebb7d4/packages/graph/hash_graph/lib/graph/src/api/rest/json_schemas/entity_type.json#L120-L123

**Review by thehabbos007 — APPROVED** [review]

**Pull request merged**